core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -163,25 +160,37 @@ object StaticFile {
       }
     } yield res)
 
-  private def ifETagModified[F[_]](req: Option[Request[F]], etagCalc: ETag) =
+  private def notModified[F[_]](
+      req: Option[Request[F]],
+      etagCalc: ETag,
+      lastModified: Option[HttpDate]): Option[Response[F]] = {
+
+    implicit val conjunction: Semigroup[Boolean] =
+      (x: Boolean, y: Boolean) => x && y
```

Comment:
This works starting in Scala 2.12 because of the single abstract method thingy, but it fails to compile in 2.12.  I think it just needs the whole `new Semigroup { ... }` ceremony.